### PR TITLE
Add cd74hc4067 docs

### PR DESCRIPTION
**Pull request by asoehlke**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#2431

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by jesserockz** [comment]
As this is a new feature matched with a PR in `esphome/esphome`, please target your PR to the `next` branch and rebase if needed.

Jesse

**Comment by asoehlke** [comment]
Hello @jesserockz, thanks, it is now targeted to next.

File: index.rst [hunk 1]
```diff
@@ -195,6 +195,7 @@ Analogue
 
     ADC, components/sensor/adc, flash.svg, ESP internal
     ADS1115, components/sensor/ads1115, ads1115.jpg, 4-channel ADC
+    CD74HC4067, components/sensor/cd74hc4067, cd74hc4067.jpg, 12-channel analog multiplexer
```
Comment (jesserockz) [inline index.rst]: Not 16 channel?
```suggestion
    CD74HC4067, components/sensor/cd74hc4067, cd74hc4067.jpg, 16-channel analog multiplexer
```

File: components/sensor/cd74hc4067.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        unit_of_measurement: "A"
+        filters:
+            - multiply: 36000
+    - platform: ct_clamp
+        name: "Solar2"
+        sensor: solar_2_raw
+        update_interval: 5s
+        unit_of_measurement: "A"
+        filters:
+            - multiply: 36000
```
Comment (jesserockz) [inline components/sensor/cd74hc4067.rst]: While I appreciate trying to help people with examples, the config block here should be specific to this component only and basically a bare minimum to get it working.
Please also remove the ct_clamp references above to make it a quicker read.

If you want to show a fuller example like this then you can make another sub-heading before `See Also` with the details.

File: components/sensor/cd74hc4067.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+Configuration variables:
+------------------------
+
+cd74hc4067 component
+********************
```
Comment (jesserockz) [inline components/sensor/cd74hc4067.rst]:
```suggestion
Component/Hub
-------------

Configuration Variables:
************************
```

File: components/sensor/cd74hc4067.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+- **pin_s0** to **pin_s3** (**Required**, :ref:`config-pin`): The I/O pins connected to the S0 to S3 channel selection pins
+- **delay** (*Optional*, :ref:`config-time`): A small delay duration needed for the chip to switch inputs, defaults to 2ms.
+- All other options from Component.
+
+cd74hc4067 sensor
+********************
```
Comment (jesserockz) [inline components/sensor/cd74hc4067.rst]:
```suggestion
Sensor
------

Configuration Variables:
************************
```

File: components/sensor/cd74hc4067.rst [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+- **sensor** (**Required**, :ref:`config-id`): The source sensor to measure voltage values from, e.g. :doc:`ADC sensor <adc>`.
+- **number** (*Required*, :ref:`config-number`): The number of the cd74hc4067 input pin (0-15)
```
Comment (jesserockz) [inline components/sensor/cd74hc4067.rst]:
```suggestion
- **number** (*Required*, int): The number of the cd74hc4067 input pin (0-15)
```

File: components/sensor/cd74hc4067.rst [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+Configuration variables:
+------------------------
+
+cd74hc4067 component
+********************
+
+- **pin_s0** to **pin_s3** (**Required**, :ref:`config-pin`): The I/O pins connected to the S0 to S3 channel selection pins
+- **delay** (*Optional*, :ref:`config-time`): A small delay duration needed for the chip to switch inputs, defaults to 2ms.
+- All other options from Component.
```
Comment (jesserockz) [inline components/sensor/cd74hc4067.rst]:
```suggestion
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/sensor/cd74hc4067.rst (hunk 2 above)
Comment (asoehlke) [reply components/sensor/cd74hc4067.rst]: The configuration is reduced to the minimum and an extra application example added.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**